live reloading breaks after I edit a markdown file 2 times (first time works somehow)

## Description

Running `gatsby develop` and editing markdown file second time is causing page component to not get any data and cause runtime errors:

### Steps to reproduce

- Clone https://github.com/NickyMeuleman/nicky-blog
- Install dependencies and run `npm run develop`
- edit `src/pages/lint-css/index.md` and save
- first edit doesn't cause errors and content is updated
- edit `src/pages/lint-css/index.md` and save again
- we get browser errors, `data` (graphql results) are not passed to page component

### Expected result

Page should be updated without errors

### Actual result

We get browser error in console:
<details>
  <summary>Browser errors</summary>
  <pre><code>
Uncaught TypeError: Cannot read property 'markdownRemark' of undefined
    at BlogPostTemplate (VM2221 blog-post.js:37)
    at ProxyComponent.hotComponentRender (D:/Projects/nicky-blog/node_modules/react-hot-loader/dist/react-hot-loader.development.js:618)
    at ProxyComponent.proxiedRender (D:/Projects/nicky-blog/node_modules/react-hot-loader/dist/react-hot-loader.development.js:635)
    at finishClassComponent (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13194)
    at updateClassComponent (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13156)
    at beginWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13825)
    at performUnitOfWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15864)
    at workLoop (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15903)
    at HTMLUnknownElement.callCallback (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:100)
    at Object.invokeGuardedCallbackDev (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:138)
    at invokeGuardedCallback (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:187)
    at replayUnitOfWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15311)
    at renderRoot (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15963)
    at performWorkOnRoot (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16561)
    at performWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16483)
    at performSyncWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16455)
    at requestWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16355)
    at scheduleWork$1 (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16219)
    at Object.enqueueSetState (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:11300)
    at ProxyComponent.Component.setState (D:/Projects/nicky-blog/node_modules/react/cjs/react.development.js:270)
    at JSONStore._this.handleMittEvent (VM2178 json-store.js:46)
    at eval (D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59)
    at Array.map (<anonymous>)
    at Object.emit (D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59)
    at r.eval (D:/Projects/nicky-blog/.cache/socketIo.js:65)
    at r.emit (index.js:83)
    at r.onevent (index.js:83)
    at r.onpacket (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.ondecoded (index.js:83)
    at a.<anonymous> (index.js:83)
    at a.r.emit (index.js:83)
    at a.add (index.js:83)
    at r.ondata (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.onPacket (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.onPacket (index.js:83)
    at r.onData (index.js:83)
    at WebSocket.ws.onmessage (index.js:83)
BlogPostTemplate @ D:/Projects/nicky-blog/src/templates/blog-post.js:37
hotComponentRender @ D:/Projects/nicky-blog/node_modules/react-hot-loader/dist/react-hot-loader.development.js:618
proxiedRender @ D:/Projects/nicky-blog/node_modules/react-hot-loader/dist/react-hot-loader.development.js:635
finishClassComponent @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13194
updateClassComponent @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13156
beginWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13825
performUnitOfWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15864
workLoop @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15903
callCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:100
invokeGuardedCallbackDev @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:138
invokeGuardedCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:187
replayUnitOfWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15311
renderRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15963
performWorkOnRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16561
performWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16483
performSyncWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16455
requestWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16355
scheduleWork$1 @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16219
enqueueSetState @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:11300
Component.setState @ D:/Projects/nicky-blog/node_modules/react/cjs/react.development.js:270
JSONStore._this.handleMittEvent @ VM2178 json-store.js:46
(anonymous) @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
emit @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
(anonymous) @ D:/Projects/nicky-blog/.cache/socketIo.js:65
r.emit @ index.js:83
r.onevent @ index.js:83
r.onpacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.ondecoded @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
a.add @ index.js:83
r.ondata @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
r.onData @ index.js:83
ws.onmessage @ index.js:83
D:/Projects/nicky-blog/node_modules/react-error-overlay/lib/index.js:2172 The above error occurred in the <BlogPostTemplate> component:
    in BlogPostTemplate (created by PageRenderer)
    in PageRenderer (at json-store.js:93)
    in JSONStore (created by RouteHandler)
    in ScrollContext (at root.js:82)
    in RouteHandler (created by Root)
    in div (created by FocusHandlerImpl)
    in FocusHandlerImpl
    in FocusHandler (created by RouterImpl)
    in RouterImpl (created by LocationProvider)
    in LocationProvider
    in Location
    in Router (created by Root)
    in Root (created by HotExportedRoot)
    in AppContainer (created by HotExportedRoot)
    in HotExportedRoot (at app.js:55)

React will try to recreate this component tree from scratch using the error boundary you provided, LocationProvider.
__stack_frame_overlay_proxy_console__ @ D:/Projects/nicky-blog/node_modules/react-error-overlay/lib/index.js:2172
__stack_frame_overlay_proxy_console__ @ D:/Projects/nicky-blog/node_modules/react-error-overlay/lib/index.js:2172
logCapturedError @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14227
logError @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14266
callback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14948
callCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:10879
commitUpdateQueue @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:10923
commitLifeCycles @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14378
commitAllLifeCycles @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15463
callCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:100
invokeGuardedCallbackDev @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:138
invokeGuardedCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:187
commitRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15604
completeRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16619
performWorkOnRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16564
performWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16483
performSyncWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16455
requestWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16355
scheduleWork$1 @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16219
enqueueSetState @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:11300
Component.setState @ D:/Projects/nicky-blog/node_modules/react/cjs/react.development.js:270
JSONStore._this.handleMittEvent @ VM2178 json-store.js:46
(anonymous) @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
emit @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
(anonymous) @ D:/Projects/nicky-blog/.cache/socketIo.js:65
r.emit @ index.js:83
r.onevent @ index.js:83
r.onpacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.ondecoded @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
a.add @ index.js:83
r.ondata @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
r.onData @ index.js:83
ws.onmessage @ index.js:83
D:/Projects/nicky-blog/node_modules/@reach/router/es/index.js:123 Uncaught TypeError: Cannot read property 'markdownRemark' of undefined
    at BlogPostTemplate (VM2221 blog-post.js:37)
    at ProxyComponent.hotComponentRender (D:/Projects/nicky-blog/node_modules/react-hot-loader/dist/react-hot-loader.development.js:618)
    at ProxyComponent.proxiedRender (D:/Projects/nicky-blog/node_modules/react-hot-loader/dist/react-hot-loader.development.js:635)
    at finishClassComponent (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13194)
    at updateClassComponent (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13156)
    at beginWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13825)
    at performUnitOfWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15864)
    at workLoop (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15903)
    at renderRoot (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15943)
    at performWorkOnRoot (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16561)
    at performWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16483)
    at performSyncWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16455)
    at requestWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16355)
    at scheduleWork$1 (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16219)
    at Object.enqueueSetState (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:11300)
    at ProxyComponent.Component.setState (D:/Projects/nicky-blog/node_modules/react/cjs/react.development.js:270)
    at JSONStore._this.handleMittEvent (VM2178 json-store.js:46)
    at eval (D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59)
    at Array.map (<anonymous>)
    at Object.emit (D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59)
    at r.eval (D:/Projects/nicky-blog/.cache/socketIo.js:65)
    at r.emit (index.js:83)
    at r.onevent (index.js:83)
    at r.onpacket (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.ondecoded (index.js:83)
    at a.<anonymous> (index.js:83)
    at a.r.emit (index.js:83)
    at a.add (index.js:83)
    at r.ondata (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.onPacket (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.onPacket (index.js:83)
    at r.onData (index.js:83)
    at WebSocket.ws.onmessage (index.js:83)
BlogPostTemplate @ D:/Projects/nicky-blog/src/templates/blog-post.js:37
hotComponentRender @ D:/Projects/nicky-blog/node_modules/react-hot-loader/dist/react-hot-loader.development.js:618
proxiedRender @ D:/Projects/nicky-blog/node_modules/react-hot-loader/dist/react-hot-loader.development.js:635
finishClassComponent @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13194
updateClassComponent @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13156
beginWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13825
performUnitOfWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15864
workLoop @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15903
renderRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15943
performWorkOnRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16561
performWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16483
performSyncWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16455
requestWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16355
scheduleWork$1 @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16219
enqueueSetState @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:11300
Component.setState @ D:/Projects/nicky-blog/node_modules/react/cjs/react.development.js:270
JSONStore._this.handleMittEvent @ VM2178 json-store.js:46
(anonymous) @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
emit @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
(anonymous) @ D:/Projects/nicky-blog/.cache/socketIo.js:65
r.emit @ index.js:83
r.onevent @ index.js:83
r.onpacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.ondecoded @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
a.add @ index.js:83
r.ondata @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
r.onData @ index.js:83
ws.onmessage @ index.js:83
D:/Projects/nicky-blog/node_modules/react-error-overlay/lib/index.js:2172 The above error occurred in the <LocationProvider> component:
    in LocationProvider
    in Location
    in Router (created by Root)
    in Root (created by HotExportedRoot)
    in AppContainer (created by HotExportedRoot)
    in HotExportedRoot (at app.js:55)

React will try to recreate this component tree from scratch using the error boundary you provided, AppContainer.
__stack_frame_overlay_proxy_console__ @ D:/Projects/nicky-blog/node_modules/react-error-overlay/lib/index.js:2172
__stack_frame_overlay_proxy_console__ @ D:/Projects/nicky-blog/node_modules/react-error-overlay/lib/index.js:2172
logCapturedError @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14227
logError @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14266
callback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14948
callCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:10879
commitUpdateQueue @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:10912
commitLifeCycles @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14378
commitAllLifeCycles @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15463
callCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:100
invokeGuardedCallbackDev @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:138
invokeGuardedCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:187
commitRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15604
completeRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16619
performWorkOnRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16564
performWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16483
performSyncWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16455
requestWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16355
scheduleWork$1 @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16219
enqueueSetState @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:11300
Component.setState @ D:/Projects/nicky-blog/node_modules/react/cjs/react.development.js:270
JSONStore._this.handleMittEvent @ VM2178 json-store.js:46
(anonymous) @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
emit @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
(anonymous) @ D:/Projects/nicky-blog/.cache/socketIo.js:65
r.emit @ index.js:83
r.onevent @ index.js:83
r.onpacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.ondecoded @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
a.add @ index.js:83
r.ondata @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
r.onData @ index.js:83
ws.onmessage @ index.js:83
D:/Projects/nicky-blog/node_modules/react-error-overlay/lib/index.js:2172 Error: A cross-origin error was thrown. React doesn't have access to the actual error object in development. See https://fb.me/react-crossorigin-error for more information.
    at Object.invokeGuardedCallbackDev (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:145)
    at invokeGuardedCallback (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:187)
    at commitRoot (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15604)
    at completeRoot (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16619)
    at performWorkOnRoot (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16564)
    at performWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16483)
    at performSyncWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16455)
    at requestWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16355)
    at scheduleWork$1 (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16219)
    at Object.enqueueSetState (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:11300)
    at ProxyComponent.Component.setState (D:/Projects/nicky-blog/node_modules/react/cjs/react.development.js:270)
    at JSONStore._this.handleMittEvent (VM2178 json-store.js:46)
    at eval (D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59)
    at Array.map (<anonymous>)
    at Object.emit (D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59)
    at r.eval (D:/Projects/nicky-blog/.cache/socketIo.js:65)
    at r.emit (index.js:83)
    at r.onevent (index.js:83)
    at r.onpacket (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.ondecoded (index.js:83)
    at a.<anonymous> (index.js:83)
    at a.r.emit (index.js:83)
    at a.add (index.js:83)
    at r.ondata (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.onPacket (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.onPacket (index.js:83)
    at r.onData (index.js:83)
    at WebSocket.ws.onmessage (index.js:83)
__stack_frame_overlay_proxy_console__ @ D:/Projects/nicky-blog/node_modules/react-error-overlay/lib/index.js:2172
__stack_frame_overlay_proxy_console__ @ D:/Projects/nicky-blog/node_modules/react-error-overlay/lib/index.js:2172
error @ D:/Projects/nicky-blog/node_modules/react-hot-loader/dist/react-hot-loader.development.js:188
componentDidCatch @ D:/Projects/nicky-blog/node_modules/react-hot-loader/dist/react-hot-loader.development.js:1567
componentDidCatch @ D:/Projects/nicky-blog/node_modules/react-hot-loader/dist/react-hot-loader.development.js:580
callback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14949
callCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:10879
commitUpdateQueue @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:10912
commitLifeCycles @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14378
commitAllLifeCycles @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15463
callCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:100
invokeGuardedCallbackDev @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:138
invokeGuardedCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:187
commitRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15604
completeRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16619
performWorkOnRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16564
performWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16483
performSyncWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16455
requestWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16355
scheduleWork$1 @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16219
enqueueSetState @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:11300
Component.setState @ D:/Projects/nicky-blog/node_modules/react/cjs/react.development.js:270
JSONStore._this.handleMittEvent @ VM2178 json-store.js:46
(anonymous) @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
emit @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
(anonymous) @ D:/Projects/nicky-blog/.cache/socketIo.js:65
r.emit @ index.js:83
r.onevent @ index.js:83
r.onpacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.ondecoded @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
a.add @ index.js:83
r.ondata @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
r.onData @ index.js:83
ws.onmessage @ index.js:83
D:/Projects/nicky-blog/src/templates/blog-post.js:37 Uncaught TypeError: Cannot read property 'markdownRemark' of undefined
    at BlogPostTemplate (VM2221 blog-post.js:37)
    at ProxyFacade (D:/Projects/nicky-blog/node_modules/react-hot-loader/dist/react-hot-loader.development.js:675)
    at mountIndeterminateComponent (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13381)
    at beginWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13821)
    at performUnitOfWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15864)
    at workLoop (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15903)
    at HTMLUnknownElement.callCallback (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:100)
    at Object.invokeGuardedCallbackDev (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:138)
    at invokeGuardedCallback (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:187)
    at replayUnitOfWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15311)
    at renderRoot (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15963)
    at performWorkOnRoot (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16561)
    at performWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16483)
    at performSyncWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16455)
    at requestWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16355)
    at scheduleWork$1 (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16219)
    at Object.enqueueSetState (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:11300)
    at ProxyComponent.Component.setState (D:/Projects/nicky-blog/node_modules/react/cjs/react.development.js:270)
    at JSONStore._this.handleMittEvent (VM2178 json-store.js:46)
    at eval (D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59)
    at Array.map (<anonymous>)
    at Object.emit (D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59)
    at r.eval (D:/Projects/nicky-blog/.cache/socketIo.js:65)
    at r.emit (index.js:83)
    at r.onevent (index.js:83)
    at r.onpacket (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.ondecoded (index.js:83)
    at a.<anonymous> (index.js:83)
    at a.r.emit (index.js:83)
    at a.add (index.js:83)
    at r.ondata (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.onPacket (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.onPacket (index.js:83)
    at r.onData (index.js:83)
    at WebSocket.ws.onmessage (index.js:83)
BlogPostTemplate @ D:/Projects/nicky-blog/src/templates/blog-post.js:37
BlogPostTemplate @ D:/Projects/nicky-blog/node_modules/react-hot-loader/dist/react-hot-loader.development.js:675
mountIndeterminateComponent @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13381
beginWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13821
performUnitOfWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15864
workLoop @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15903
callCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:100
invokeGuardedCallbackDev @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:138
invokeGuardedCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:187
replayUnitOfWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15311
renderRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15963
performWorkOnRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16561
performWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16483
performSyncWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16455
requestWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16355
scheduleWork$1 @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16219
enqueueSetState @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:11300
Component.setState @ D:/Projects/nicky-blog/node_modules/react/cjs/react.development.js:270
JSONStore._this.handleMittEvent @ VM2178 json-store.js:46
(anonymous) @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
emit @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
(anonymous) @ D:/Projects/nicky-blog/.cache/socketIo.js:65
r.emit @ index.js:83
r.onevent @ index.js:83
r.onpacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.ondecoded @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
a.add @ index.js:83
r.ondata @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
r.onData @ index.js:83
ws.onmessage @ index.js:83
D:/Projects/nicky-blog/node_modules/react-error-overlay/lib/index.js:2172 The above error occurred in the <BlogPostTemplate> component:
    in BlogPostTemplate (created by PageRenderer)
    in PageRenderer (at json-store.js:93)
    in JSONStore (created by RouteHandler)
    in ScrollContext (at root.js:82)
    in RouteHandler (created by Root)
    in div (created by FocusHandlerImpl)
    in FocusHandlerImpl
    in FocusHandler (created by RouterImpl)
    in RouterImpl (created by LocationProvider)
    in LocationProvider
    in Location
    in Router (created by Root)
    in Root (created by HotExportedRoot)
    in AppContainer (created by HotExportedRoot)
    in HotExportedRoot (at app.js:55)

React will try to recreate this component tree from scratch using the error boundary you provided, LocationProvider.
__stack_frame_overlay_proxy_console__ @ D:/Projects/nicky-blog/node_modules/react-error-overlay/lib/index.js:2172
__stack_frame_overlay_proxy_console__ @ D:/Projects/nicky-blog/node_modules/react-error-overlay/lib/index.js:2172
logCapturedError @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14227
logError @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14266
callback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14948
callCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:10879
commitUpdateQueue @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:10923
commitLifeCycles @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14378
commitAllLifeCycles @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15463
callCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:100
invokeGuardedCallbackDev @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:138
invokeGuardedCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:187
commitRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15604
completeRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16619
performWorkOnRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16564
performWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16483
performSyncWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16455
requestWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16355
scheduleWork$1 @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16219
enqueueSetState @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:11300
Component.setState @ D:/Projects/nicky-blog/node_modules/react/cjs/react.development.js:270
JSONStore._this.handleMittEvent @ VM2178 json-store.js:46
(anonymous) @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
emit @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
(anonymous) @ D:/Projects/nicky-blog/.cache/socketIo.js:65
r.emit @ index.js:83
r.onevent @ index.js:83
r.onpacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.ondecoded @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
a.add @ index.js:83
r.ondata @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
r.onData @ index.js:83
ws.onmessage @ index.js:83
D:/Projects/nicky-blog/node_modules/@reach/router/es/index.js:123 Uncaught TypeError: Cannot read property 'markdownRemark' of undefined
    at BlogPostTemplate (VM2221 blog-post.js:37)
    at ProxyFacade (D:/Projects/nicky-blog/node_modules/react-hot-loader/dist/react-hot-loader.development.js:675)
    at mountIndeterminateComponent (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13381)
    at beginWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13821)
    at performUnitOfWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15864)
    at workLoop (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15903)
    at renderRoot (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15943)
    at performWorkOnRoot (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16561)
    at performWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16483)
    at performSyncWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16455)
    at requestWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16355)
    at scheduleWork$1 (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16219)
    at Object.enqueueSetState (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:11300)
    at ProxyComponent.Component.setState (D:/Projects/nicky-blog/node_modules/react/cjs/react.development.js:270)
    at JSONStore._this.handleMittEvent (VM2178 json-store.js:46)
    at eval (D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59)
    at Array.map (<anonymous>)
    at Object.emit (D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59)
    at r.eval (D:/Projects/nicky-blog/.cache/socketIo.js:65)
    at r.emit (index.js:83)
    at r.onevent (index.js:83)
    at r.onpacket (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.ondecoded (index.js:83)
    at a.<anonymous> (index.js:83)
    at a.r.emit (index.js:83)
    at a.add (index.js:83)
    at r.ondata (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.onPacket (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.onPacket (index.js:83)
    at r.onData (index.js:83)
    at WebSocket.ws.onmessage (index.js:83)
BlogPostTemplate @ D:/Projects/nicky-blog/src/templates/blog-post.js:37
BlogPostTemplate @ D:/Projects/nicky-blog/node_modules/react-hot-loader/dist/react-hot-loader.development.js:675
mountIndeterminateComponent @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13381
beginWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:13821
performUnitOfWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15864
workLoop @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15903
renderRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15943
performWorkOnRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16561
performWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16483
performSyncWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16455
requestWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16355
scheduleWork$1 @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16219
enqueueSetState @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:11300
Component.setState @ D:/Projects/nicky-blog/node_modules/react/cjs/react.development.js:270
JSONStore._this.handleMittEvent @ VM2178 json-store.js:46
(anonymous) @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
emit @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
(anonymous) @ D:/Projects/nicky-blog/.cache/socketIo.js:65
r.emit @ index.js:83
r.onevent @ index.js:83
r.onpacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.ondecoded @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
a.add @ index.js:83
r.ondata @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
r.onData @ index.js:83
ws.onmessage @ index.js:83
D:/Projects/nicky-blog/node_modules/react-error-overlay/lib/index.js:2172 The above error occurred in the <LocationProvider> component:
    in LocationProvider
    in Location
    in Router (created by Root)
    in Root (created by HotExportedRoot)
    in AppContainer (created by HotExportedRoot)
    in HotExportedRoot (at app.js:55)

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
__stack_frame_overlay_proxy_console__ @ D:/Projects/nicky-blog/node_modules/react-error-overlay/lib/index.js:2172
__stack_frame_overlay_proxy_console__ @ D:/Projects/nicky-blog/node_modules/react-error-overlay/lib/index.js:2172
logCapturedError @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14227
logError @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14266
update.callback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14919
callCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:10879
commitUpdateQueue @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:10912
commitLifeCycles @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:14397
commitAllLifeCycles @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15463
callCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:100
invokeGuardedCallbackDev @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:138
invokeGuardedCallback @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:187
commitRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15604
completeRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16619
performWorkOnRoot @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16564
performWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16483
performSyncWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16455
requestWork @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16355
scheduleWork$1 @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16219
enqueueSetState @ D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:11300
Component.setState @ D:/Projects/nicky-blog/node_modules/react/cjs/react.development.js:270
JSONStore._this.handleMittEvent @ VM2178 json-store.js:46
(anonymous) @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
emit @ D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59
(anonymous) @ D:/Projects/nicky-blog/.cache/socketIo.js:65
r.emit @ index.js:83
r.onevent @ index.js:83
r.onpacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.ondecoded @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
a.add @ index.js:83
r.ondata @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
r.onData @ index.js:83
ws.onmessage @ index.js:83
D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:145 Uncaught Error: A cross-origin error was thrown. React doesn't have access to the actual error object in development. See https://fb.me/react-crossorigin-error for more information.
    at Object.invokeGuardedCallbackDev (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:145)
    at invokeGuardedCallback (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:187)
    at commitRoot (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:15604)
    at completeRoot (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16619)
    at performWorkOnRoot (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16564)
    at performWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16483)
    at performSyncWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16455)
    at requestWork (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16355)
    at scheduleWork$1 (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:16219)
    at Object.enqueueSetState (D:/Projects/nicky-blog/node_modules/react-dom/cjs/react-dom.development.js:11300)
    at ProxyComponent.Component.setState (D:/Projects/nicky-blog/node_modules/react/cjs/react.development.js:270)
    at JSONStore._this.handleMittEvent (VM2178 json-store.js:46)
    at eval (D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59)
    at Array.map (<anonymous>)
    at Object.emit (D:/Projects/nicky-blog/node_modules/mitt/dist/mitt.es.js:59)
    at r.eval (D:/Projects/nicky-blog/.cache/socketIo.js:65)
    at r.emit (index.js:83)
    at r.onevent (index.js:83)
    at r.onpacket (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.ondecoded (index.js:83)
    at a.<anonymous> (index.js:83)
    at a.r.emit (index.js:83)
    at a.add (index.js:83)
    at r.ondata (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.onPacket (index.js:83)
    at r.<anonymous> (index.js:83)
    at r.emit (index.js:83)
    at r.onPacket (index.js:83)
    at r.onData (index.js:83)
    at WebSocket.ws.onmessage (index.js:83)
  </code></pre>
</details>

### Environment

```
  System:
    OS: Windows 10
    CPU: x64 Intel(R) Core(TM) i5-3210M CPU @ 2.50GHz
  Binaries:
    Yarn: 1.7.0 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.1.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: 42.17134.1.0
  npmPackages:
    gatsby: ^2.0.0-beta.100 => 2.0.0-beta.100
    gatsby-image: next => 2.0.0-beta.7
    gatsby-plugin-catch-links: ^2.0.2-beta.8 => 2.0.2-beta.8
    gatsby-plugin-extract-schema: 0.0.4 => 0.0.4
    gatsby-plugin-google-analytics: ^2.0.0-beta.5 => 2.0.0-beta.5
    gatsby-plugin-manifest: ^2.0.2-beta.5 => 2.0.2-beta.5
    gatsby-plugin-netlify: ^2.0.0-beta.6 => 2.0.0-beta.6
    gatsby-plugin-netlify-cache: ^0.1.0 => 0.1.0
    gatsby-plugin-offline: ^2.0.0-beta.9 => 2.0.0-beta.9
    gatsby-plugin-react-helmet: next => 3.0.0-beta.4
    gatsby-plugin-remove-trailing-slashes: ^1.0.9 => 1.0.9
    gatsby-plugin-sharp: next => 2.0.0-beta.7
    gatsby-plugin-sitemap: ^2.0.0-beta.4 => 2.0.0-beta.4
    gatsby-plugin-styled-components: next => 3.0.0-beta.3
    gatsby-plugin-typography: next => 2.2.0-beta.3
    gatsby-remark-autolink-headers: ^2.0.0-beta.5 => 2.0.0-beta.5
    gatsby-remark-copy-linked-files: next => 2.0.0-beta.3
    gatsby-remark-images: ^2.0.1-beta.10 => 2.0.1-beta.10
    gatsby-remark-prismjs: next => 3.0.0-beta.5
    gatsby-source-filesystem: next => 2.0.1-beta.10
    gatsby-transformer-remark: ^2.1.1-beta.6 => 2.1.1-beta.6
    gatsby-transformer-sharp: ^2.1.1-beta.7 => 2.1.1-beta.7
```

### File contents (if changed)

`gatsby-config.js`: 
```
module.exports = {
  siteMetadata: {
    title: 'Nicky blogs',
    siteUrl: 'https://nickymeuleman.netlify.com',
  },
  plugins: [
    'gatsby-plugin-react-helmet',
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `pages`,
        path: `${__dirname}/src/pages/`,
      },
    },
    `gatsby-transformer-sharp`,
    `gatsby-plugin-sharp`,
    {
      resolve: `gatsby-transformer-remark`,
      options: {
        plugins: [
          {
            resolve: `gatsby-remark-images`,
            options: {
              maxWidth: 590,
              linkImagesToOriginal: false,
            },
          },
          {
            resolve: `gatsby-remark-prismjs`,
            options: {
              classPrefix: 'language-',
              inlineCodeMarker: null,
              aliases: {},
            },
          },
          'gatsby-remark-copy-linked-files',
          `gatsby-remark-autolink-headers`,
        ],
      },
    },
    `gatsby-plugin-catch-links`,
    `gatsby-plugin-styled-components`,
    {
      resolve: `gatsby-plugin-typography`,
      options: {
        pathToConfigModule: `src/utils/typography.js`,
        omitGoogleFont: true,
      },
    },
    {
      resolve: `gatsby-plugin-manifest`,
      options: {
        name: 'Nicky Meuleman',
        short_name: 'NickyM',
        start_url: '/',
        background_color: '#f5f5f5',
        theme_color: '#155799',
        display: 'minimal-ui',
        icon: 'static/favicon.svg',
      },
    },
    'gatsby-plugin-offline',
    {
      resolve: `gatsby-plugin-sitemap`,
    },
    {
      resolve: `gatsby-plugin-google-analytics`,
      options: {
        trackingId: 'UA-117805994-1',
        head: false,
        anonymize: true,
        respectDNT: true,
      },
    },
    `gatsby-plugin-remove-trailing-slashes`,
    'gatsby-plugin-netlify-cache',
    'gatsby-plugin-extract-schema',
    'gatsby-plugin-netlify', // keep as last in array
  ],
};
```
`package.json`: 
```
{
  "name": "nicky-blog",
  "description": "Nicky tries to blog",
  "version": "0.2.0",
  "author": "Nicky Meuleman",
  "repository": {
    "type": "git",
    "url": "https://github.com/NickyMeuleman/nicky-blog.git"
  },
  "bugs": {
    "url": "https://github.com/NickyMeuleman/nicky-blog/issues"
  },
  "keywords": [
    "blog",
    "gatsby",
    "web development"
  ],
  "homepage": "https://nickymeuleman.netlify.com/",
  "dependencies": {
    "babel-plugin-styled-components": "^1.5.1",
    "gatsby": "^2.0.0-beta.100",
    "gatsby-image": "next",
    "gatsby-plugin-catch-links": "^2.0.2-beta.8",
    "gatsby-plugin-extract-schema": "0.0.4",
    "gatsby-plugin-google-analytics": "^2.0.0-beta.5",
    "gatsby-plugin-manifest": "^2.0.2-beta.5",
    "gatsby-plugin-netlify": "^2.0.0-beta.6",
    "gatsby-plugin-netlify-cache": "^0.1.0",
    "gatsby-plugin-offline": "^2.0.0-beta.9",
    "gatsby-plugin-react-helmet": "next",
    "gatsby-plugin-sharp": "next",
    "gatsby-plugin-sitemap": "^2.0.0-beta.4",
    "gatsby-plugin-styled-components": "next",
    "gatsby-plugin-typography": "next",
    "gatsby-remark-autolink-headers": "^2.0.0-beta.5",
    "gatsby-remark-copy-linked-files": "next",
    "gatsby-remark-images": "^2.0.1-beta.10",
    "gatsby-remark-prismjs": "next",
    "gatsby-source-filesystem": "next",
    "gatsby-transformer-remark": "^2.1.1-beta.6",
    "gatsby-transformer-sharp": "^2.1.1-beta.7",
    "prismjs": "^1.15.0",
    "react": "^16.4.2",
    "react-dom": "^16.4.2",
    "react-helmet": "^5.2.0",
    "react-typography": "^0.16.13",
    "styled-components": "^3.4.2",
    "typed.js": "^2.0.9",
    "typeface-roboto": "0.0.54",
    "typeface-roboto-slab": "0.0.54",
    "typography": "^0.16.17",
    "typography-breakpoint-constants": "^0.15.10",
    "typography-theme-ocean-beach": "^0.15.11"
  },
  "license": "GPL-3.0-or-later",
  "scripts": {
    "build": "gatsby build",
    "serve": "gatsby serve",
    "dev": "gatsby develop",
    "format": "prettier --write 'src/**/*.js'",
    "test": "echo \"Error: no test specified\" && exit 1",
    "commit": "commit"
  },
  "husky": {
    "hooks": {
      "pre-commit": "lint-staged",
      "commit-msg": "commitlint -e %HUSKY_GIT_PARAMS%"
    }
  },
  "lint-staged": {
    "*.{js,jsx}": [
      "eslint --fix",
      "git add"
    ],
    "*.{json,css}": [
      "prettier --write",
      "git add"
    ]
  },
  "devDependencies": {
    "@commitlint/cli": "^6.2.0",
    "@commitlint/config-conventional": "^6.1.3",
    "@commitlint/prompt-cli": "^6.1.3",
    "babel-eslint": "^8.2.5",
    "eslint": "^4.19.1",
    "eslint-config-airbnb": "^16.1.0",
    "eslint-config-prettier": "^2.10.0",
    "eslint-plugin-graphql": "^2.1.1",
    "eslint-plugin-import": "^2.14.0",
    "eslint-plugin-jsx-a11y": "^6.0.3",
    "eslint-plugin-prettier": "^2.6.0",
    "eslint-plugin-react": "^7.11.0",
    "gatsby-plugin-remove-trailing-slashes": "^1.0.9",
    "husky": "^1.0.0-rc.9",
    "lint-staged": "^7.2.2",
    "prettier": "^1.14.2"
  }
}
```
`gatsby-node.js`:
```js
const path = require(`path`);
const { createFilePath } = require(`gatsby-source-filesystem`);

exports.onCreateNode = ({ node, getNode, actions }) => {
  const { createNodeField } = actions;
  if (node.internal.type === 'MarkdownRemark') {
    const slug = createFilePath({ node, getNode, basePath: 'pages' });
    createNodeField({
      node,
      name: 'slug',
      value: slug,
    });
  }
};

exports.createPages = ({ graphql, actions }) => {
  const { createPage } = actions;
  return new Promise((resolve, reject) => {
    graphql(
      `
        {
          allMarkdownRemark {
            edges {
              node {
                fields {
                  slug
                }
              }
            }
          }
        }
      `
    ).then(result => {
      result.data.allMarkdownRemark.edges.forEach(({ node }) => {
        createPage({
          path: `/blog${node.fields.slug}`,
          component: path.resolve('./src/templates/blog-post.js'),
          context: {
            slug: node.fields.slug,
          },
        });
      });
      resolve();
    });
  });
};
```
`gatsby-browser.js`: N/A <!-- Please use a code block or just leave it as is if wasn't changed -->
`gatsby-ssr.js`: N/A <!-- Please use a code block or just leave it as is if wasn't changed -->

--- a/packages/gatsby/src/internal-plugins/query-runner/query-watcher.js
+++ b/packages/gatsby/src/internal-plugins/query-runner/query-watcher.js
@@ -13,7 +13,7 @@ const chokidar = require(`chokidar`)
 const path = require(`path`)
 const slash = require(`slash`)
 
-const { store } = require(`../../redux/`)
+const { store, emitter } = require(`../../redux/`)
 const { boundActionCreators } = require(`../../redux/actions`)
 const queryCompiler = require(`./query-compiler`).default
 const report = require(`gatsby-cli/lib/reporter`)
@@ -247,4 +247,32 @@ const watch = rootDir => {
       debounceCompile()
     })
   filesToWatch.forEach(filePath => watcher.add(filePath))
+}
+
+if (process.env.gatsby_executing_command === `develop`) {
+  let bootstrapFinished = false
+  emitter.on(`BOOTSTRAP_FINISHED`, () => {
+    bootstrapFinished = true
+  })
+  emitter.on(`DELETE_PAGE`, action => {
+    if (bootstrapFinished) {
+      const componentPath = slash(action.payload.component)
+      const { pages } = store.getState()
+      let otherPageWithTemplateExists = false
+      for (let page of pages.values()) {
+        if (slash(page.component) === componentPath) {
+          otherPageWithTemplateExists = true
+          break
+        }
+      }
+      if (!otherPageWithTemplateExists) {
+        store.dispatch({
+          type: `REMOVE_TEMPLATE_COMPONENT`,
+          payload: {
+            componentPath,
+          },
+        })
+      }
+    }
+  })
 }

--- a/packages/gatsby/src/redux/reducers/components.js
+++ b/packages/gatsby/src/redux/reducers/components.js
@@ -14,10 +14,6 @@ module.exports = (state = new Map(), action) => {
         })
       )
       return state
-    case `DELETE_PAGE`:
-      action.payload.componentPath = normalize(action.payload.component)
-      state.delete(action.payload.componentPath)
-      return state
     case `REMOVE_TEMPLATE_COMPONENT`:
       state.delete(normalize(action.payload.componentPath))
       return state

